### DISCUSSION TITLE
Export runner logs to DataDog.

### DISCUSSION THREAD

**Discussion by janani2019** [actions/runner#3529] [Q&A]
Hi,
We are hosted Self-hosted-runners on EKS cluster. 
I see that workflow action logs are available in /_diag/pages folder. Tried to mount the folder onto the host path, but strangely, only the empty files get mounted. But, if I create a file with content in the same folder, that gets mounted.  Is there any difference in the files mounted with logs?  also Please suggest ways to export the logs to datadog as the pods are ephemeral.  

from runner:
<img width="934" alt="Screenshot 2024-10-25 at 11 32 19 AM" src="https://github.com/user-attachments/assets/5ef3f254-a96b-4ece-b23a-e22e4899398d">

From host (hostPath):

![Screenshot 2024-10-25 at 11 33 22 AM](https://github.com/user-attachments/assets/a43695fa-4ee5-41ad-912c-a48644f1c7ef)

**Comment by JeetKarena** [comment]
GitHub runner logs in the `_diag` directory may be created by the runner process and not flushed immediately, or they may be opened with special access modes that don’t ensure immediate content availability over a volume mount. Content written by the runner could be delayed, virtualized, or flushed only at process exit, especially if the process uses various buffering or locking mechanisms.

All log files within the `_diag` directory can be imported by Datadog, provided the agent configuration is set up to tail every relevant file type (such as logs, pages, and any specific diagnostic outputs). There is no technical limitation preventing Datadog from collecting multiple log formats, and the agent’s configuration can include glob patterns that match all files, for example: `/actions-runner/_diag/*.*`
